metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+
+           SegmentStats.countBranchPoints - nodes with more than 2 nodes connected to them
+
+           SegmentStats.contractionDict - A dictionary with key as the (branching index (nth branch from the start node), start node,
+                                                end node of the branch) value = contraction of the branch
+
+           SegmentStats.hausdorffDimensionDict - A dictionary with key as the (branching index (nth branch from the start node), start node,
+                                                       end node of the branch) value = hausdorff dimension of the branch
+
+
+    Notes
+    --------
+
+    tortuosity = curveLength / curveDisplacement
+    contraction = curveDisplacement / curveLength (better becuase there is no change of instability (undefined) in case of cycles)
+    Hausdorff Dimension = np.log(curveLength) / np.log(curveDisplacement) https://en.wikipedia.org/wiki/Hausdorff_dimension
+    Type of subgraphs:
+    0 = if graph a single node
+    1 = if graph is a single cycle
+    2 = line (highest degree in the subgraph is 2)
+    3 = undirected acyclic graph
+    4 = undirected cyclic graph
+
+    """
+    def __init__(self, networkxGraph):
+        self.networkxGraph = networkxGraph
+        # intitialize all the instance variables of SegmentStats class
+        self.contractionDict = {}
+        self.countDict = {}
+        self.cycleInfoDict = {}
+        self.hausdorffDimensionDict = {}
+        self.isolatedEdgeInfoDict = {}
+        self.lengthDict = {}
+        self.tortuosityDict = {}
+        self.typeGraphdict = {}
+
+        self.cycles = 0
+        self.edgesUntraced = 0
+        self.totalSegments = 0
+        self._sortedSegments = []
+        self._visitedSources = []
+        self._visitedPaths = []
+        # list of _disjointGraphs
+        self._disjointGraphs = list(nx.connected_component_subgraphs(self.networkxGraph))
+
+    def _getLengthAndRemoveTracedPath(self, subGraphSkeleton, path, isCycle=0, remove=1):
+        """
+        Find length of a path as distance between nodes in it
+        and Remove edges belonging to "path" from "subGraphSkeleton"
+        Parameters
+        ----------
+        subGraphSkeleton : Networkx graph
+
+        path : list
+           list of nodes in the path
+
+        isCycle : boolean
+           Specify if path is a cycle or not
+
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited edges removed
+
+        length : float
+            Length of path
+
+        Notes
+        ------
+        given a visited path in variable path, the edges in the
+        path are removed in the graph
+        if isCycle = 1 , the given path belongs to a cycle,
+        so an additional edge is formed between the last and the
+        first node to form a closed cycle/ path and is removed
+        and remove the edges
```

Comment:
I'm not sure what this last line means. In general, multiple-sentence comments should begin with capital letters and have punctuation. It's a little nitpicking, but I think it's worth it long-term.